### PR TITLE
Custom Workspace Tab

### LINKED ISSUE #16693: [new UD] Implement "Create Custom Workspace" page
### Is your task related to a problem? Please describe.

We need the "Create Custom Workspace" option on "Getting Started". It should be the same design and functionality of the page which we have for the existing Dashboard.

### Describe the solution you'd like

Screenshot from 2020-04-21 19-01-27

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

Implement Custom Workspace Tab.

Screenshot 2020-06-25 at 19 13 01

### What issue does this PR address?

resolves https://github.com/eclipse/che/issues/16693

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -26,21 +27,22 @@ type RouteItem = {
...
-const GetStartedPage = React.lazy(() => import('./app-nav-menu/get-started/GetStartedPage'));
+const GetStartedPage = React.lazy(() => import('./app-nav-menu/getting-started/GetStartedPage'));
 const WorkspacesList = React.lazy(() => import('./app-nav-menu/workspaces/WorkspacesList'));
 const Administration = React.lazy(() => import('./app-nav-menu/administration/Administration'));
 const WorkspaceDetails = React.lazy(() => import('./workspace-details/WorkspaceDetails'));
 const IdeIframe = React.lazy(() => import('./ide-iframe/IdeIframe'));
 
 const items: RouteItem[] = [
-  { to: '/', component: GetStartedPage, label: 'Get Started Page', ico: 'fa fa-plus' },
+  { to: '/get-started', component: GetStartedPage, label: 'Get Started Page', ico: 'fa fa-plus' },
```
Comment (olexii4) [inline src/components/App.tsx:37]: /get-started  ->  getting-started (like './app-nav-menu/getting-started/GetStartedPage')

Comment (olexii4) [reply src/components/App.tsx:37]: WDYT?

File: src/components/app-nav-menu/getting-started/GetStartedPage.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
+/*
```
Comment (olexii4) [inline src/components/app-nav-menu/getting-started/GetStartedPage.tsx]: What about renaming (GetStartedPage.tsx -> get-started-page.tsx or something like this) ?

File: src/components/app-nav-menu/getting-started/custom-workspace-tab/CustomWorkspaceTab.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+//   & { history: any };
+
+// type State = {
+//   alertVisible: boolean;
+//   temporary: boolean;
+// };
+
+export class CustomWorkspaceTab extends React.Component {
+
+  constructor(props: any) {
```
Comment (olexii4) [inline src/components/app-nav-menu/getting-started/custom-workspace-tab/CustomWorkspaceTab.tsx]: Could we avoid 'any' in this case?

File: src/components/app-nav-menu/getting-started/custom-workspace-tab/InfrastructureNamespace.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
+/*
```
Comment (olexii4) [inline src/components/app-nav-menu/getting-started/custom-workspace-tab/InfrastructureNamespace.tsx]: InfrastructureNamespace.tsx -> infrastructure-namespace.tsx  ?

File: src/components/app-nav-menu/getting-started/custom-workspace-tab/InfrastructureNamespaceSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
+/*
```
Comment (olexii4) [inline src/components/app-nav-menu/getting-started/custom-workspace-tab/InfrastructureNamespaceSelect.tsx]: InfrastructureNamespaceSelect.tsx -> infrastructure-namespace-select.tsx ?

File: src/components/app-nav-menu/getting-started/custom-workspace-tab/CustomWorkspaceTab.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        <span>Workspace Name</span>
+        <span>Temporary Storage</span>
+        <span>Devfile selector</span>
+        <span>Devfile editor</span>
+      </React.Fragment>
+    );
+  }
+}
+
+export default connect(
```
Comment (olexii4) [inline src/components/app-nav-menu/getting-started/custom-workspace-tab/CustomWorkspaceTab.tsx]: Do we really need this 'connect' ?

Comment (akurinnoy) [reply src/components/app-nav-menu/getting-started/custom-workspace-tab/CustomWorkspaceTab.tsx]: @olexii4 please pay attention that this is a draft PR and it's still in progress.

File: src/components/app-nav-menu/getting-started/custom-workspace-tab/TemporaryStorage.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  public render(): React.ReactNode {
+    const { isTemporary } = this.state;
+    return (
+      <FormGroup
+        label="Temporary Storage"
+        labelIcon={
+          <Tooltip
+            isContentLeftAligned
+            content={
+              <div>Temporary storage allows for faster I/O but may have limited storage and is not persistent.</div>
```
Comment (olexii4) [inline src/components/app-nav-menu/getting-started/custom-workspace-tab/TemporaryStorage.tsx]:
```
const  tooltipContent = (
    <div>
        Temporary storage allows for faster I/O but may have limited storage and is not persistent.
    </div>
);
```
WDYT?

File: src/components/App.tsx (hunk 1 above)
Comment (akurinnoy) [reply src/components/App.tsx:37]: I reverter that renaming.

File: src/components/app-nav-menu/getting-started/GetStartedPage.tsx (hunk 2 above)
Comment (akurinnoy) [reply src/components/app-nav-menu/getting-started/GetStartedPage.tsx]: Please open an issue to agree file naming convention.

**Review by sleshchenko — COMMENTED** [review]
I haven't tested it but the provided screenshot LGTM :+1:

**Review by olexii4 — APPROVED** [review]

**Pull request merged**